src/SignalR/server/Core/src/HubConnectionContext.cs

```diff
@@ -69,7 +70,13 @@ public HubConnectionContext(ConnectionContext connectionContext, HubConnectionCo
             _connectionContext = connectionContext;
             _logger = loggerFactory.CreateLogger<HubConnectionContext>();
             ConnectionAborted = _connectionAbortedTokenSource.Token;
-            _closedRegistration = connectionContext.ConnectionClosed.Register((state) => ((HubConnectionContext)state!).Abort(), this);
+            _closedRegistration = connectionContext.ConnectionClosed.Register(static (state) => ((HubConnectionContext)state!).Abort(), this);
+
+            if (connectionContext.Features.Get<IConnectionLifetimeNotificationFeature>() is IConnectionLifetimeNotificationFeature lifetimeNotification)
+            {
+                // This feature is used by HttpConnectionManager to gracefully close the connection on authentication expiration
```

Comment:
It seems weird to me to call it graceful when we're calling abort:

```suggestion
                // This feature is used by HttpConnectionManager to close the connection with a non-errored close message.
```

Or should we include an error in the close message indicating that auth expired?